config/config.go

```diff
@@ -224,16 +224,28 @@ type P2PConfig struct {
 
 	// Time to wait before flushing messages out on the connection. In ms
 	FlushThrottleTimeout int `mapstructure:"flush_throttle_timeout"`
+
+	// Maximum size of a message packet payload
+	MaxMsgPacketPayloadSize int `mapstructure:"max_msg_packet_payload_size"`
+
+	// Rate at which packets can be sent (in bytes/second)
+	SendRate int64 `mapstructure:"send_rate"`
+
+	// Rate at which packets can be received (in bytes/second)
+	RecvRate int64 `mapstructure:"recv_rate"`
 }
 
 // DefaultP2PConfig returns a default configuration for the peer-to-peer layer
 func DefaultP2PConfig() *P2PConfig {
 	return &P2PConfig{
-		ListenAddress:        "tcp://0.0.0.0:46656",
-		AddrBook:             "addrbook.json",
-		AddrBookStrict:       true,
-		MaxNumPeers:          50,
-		FlushThrottleTimeout: 100,
+		ListenAddress:           "tcp://0.0.0.0:46656",
+		AddrBook:                "addrbook.json",
+		AddrBookStrict:          true,
+		MaxNumPeers:             50,
+		FlushThrottleTimeout:    100,
+		MaxMsgPacketPayloadSize: 1024,   // 1 kB
+		SendRate:                512000, // 500 kB/s
```

Comment:
Minor nit, could we use the defaults in `p2p/connection.go`

Reply:
We could, but we'd have circular imports.

We do import `types` to use a default value, but I really don't like that either. Ideally we clear the default values out of `p2p` soon and put them all here. Does that sound right?

Reply:
Oh crap, my bad, for some reason it seemed like they were in the same package, thanks for letting me know. The only thing left is the commit message request then my LGTM is in.